yt_dlp/extractor/youtube.py

```diff
@@ -941,54 +941,57 @@ def _parse_time_text(self, text):
     def _extract_response(self, item_id, query, note='Downloading API JSON', headers=None,
                           ytcfg=None, check_get_keys=None, ep='browse', fatal=True, api_hostname=None,
                           default_client='web'):
-        for retry in self.RetryManager():
-            try:
-                response = self._call_api(
-                    ep=ep, fatal=True, headers=headers,
-                    video_id=item_id, query=query, note=note,
-                    context=self._extract_context(ytcfg, default_client),
-                    api_key=self._extract_api_key(ytcfg, default_client),
-                    api_hostname=api_hostname, default_client=default_client)
-            except ExtractorError as e:
-                if not isinstance(e.cause, network_exceptions):
-                    return self._error_or_warning(e, fatal=fatal)
-                elif not isinstance(e.cause, HTTPError):
-                    retry.error = e
-                    continue
+        for icd_retry in self.RetryManager(
+            fatal=self._configuration_arg('raise_incomplete_data', [False], ie_key=YoutubeIE)[0] is not False
+        ):
```

Comment:
Why is this the outer one? Shouldn't we have X retries overall, and each of those can use X incomplete data?

Reply:
Hmm, you might have a point. I'll have a think